front/app/api/files/filePreviews/useFilePreview.ts

```diff
@@ -0,0 +1,27 @@
+import { useQuery } from '@tanstack/react-query';
+import { CLErrors } from 'typings';
+
+import fetcher from 'utils/cl-react-query/fetcher';
+
+import filePreviewsKeys from './keys';
+import { IFilePreview, FilePreviewKeys } from './types';
+
+const fetchPreviewByFileId = (id?: string | null) =>
+  fetcher<IFilePreview>({
+    path: `/files/${id}/preview`,
+    action: 'get',
+  });
+
+const useFilePreview = (id?: string | null) => {
```

Comment:
Not huge problem but does this type really need to be `string` | `null` | `undefined`? Is `string` | `undefined` also ok? Otherwise it might create the impression to someone using the function that `undefined` and `null` have a different behavior or something